apps/net/httpclient/src/main.rs

```diff
@@ -4,27 +4,71 @@
 #[macro_use]
 extern crate libax;
 
-use core::str::FromStr;
+extern crate alloc;
 
+use alloc::string::{String, ToString};
+use alloc::vec::Vec;
 use libax::io::{self, prelude::*};
-use libax::net::{IpAddr, TcpStream};
+use libax::net::{SocketAddr, TcpStream, ToSocketAddrs};
 
-const DEST_IP: &str = "49.12.234.183"; // ident.me
-const REQUEST: &str = "\
-GET / HTTP/1.1\r\n\
-Host: ident.me\r\n\
-Accept: */*\r\n\
-\r\n";
+const DEST_HOST: &str = "www.example.com";
+const DEST_IP: &str = "93.184.216.34";
+
+fn get_addr() -> SocketAddr {
+    let dest = if cfg!(feature = "dns") {
+        print!("{} ", DEST_HOST);
+        DEST_HOST
+    } else {
+        DEST_IP
+    };
+    let addr_iter = (dest, 80).to_socket_addrs().unwrap().collect::<Vec<_>>();
+    println!("IP:{}\n", addr_iter[0].addr);
+    addr_iter[0]
+}
 
 fn client() -> io::Result {
-    let (addr, port) = (IpAddr::from_str(DEST_IP).unwrap(), 80);
-    let mut stream = TcpStream::connect((addr, port).into())?;
-    stream.write(REQUEST.as_bytes())?;
-
-    let mut buf = [0; 1024];
-    let n = stream.read(&mut buf)?;
-    let response = core::str::from_utf8(&buf[..n]).unwrap();
-    println!("{}", response);
+    let requset: String =
+        "GET / HTTP/1.1\r\nHost: ".to_string() + DEST_HOST + "\r\nAccept: */*\r\n\r\n";
+
+    let mut stream = TcpStream::connect(get_addr())?;
+    stream.write(requset.as_bytes())?;
+    let mut buf = [0; 2048];
+    let mut length;
+    {
```

Comment:
I think the original version is fine, this code leads to the example is not simple enough.